### DISCUSSION TITLE
Partial diffusion

### DISCUSSION THREAD

**Discussion by ulysse06** [ProteinDesignLab/protpardelle#2] [Q&A] [answered]
Amazing project!

I would like to apply this tool for my research but I couldn't find an easy way to perform controlled partial diffusion with protpardelle. (I have been doing it with [RFdiffusion](https://github.com/RosettaCommons/RFdiffusion) until now)
Is it something that could be achieved or that you would eventually consider implementing?

**Comment by alexechu** [comment, marked as answer]
Thanks! You can definitely perform "partial diffusion" with protpardelle - for now as a hack, you could probably use the `model.sample()` function (with the default sampling configs) to do this: set the `gt_coords` and `gt_cond_atom_mask` to your input coordinates, and set `apply_cond_proportion=0.8` for example to only generate for the last 20% of the reverse diffusion. You can do this with the aatype as well to fix the sequence up until the last 20%. We'll definitely look to add this as a more convenient feature in future updates.
